developing/docker/README.md

```diff
@@ -94,6 +98,162 @@ To get a `top` like report of what your containers are doing
 
     docker stats
 
+## Image IDs, Tags, and Digests
+
+Docker image identifiers and SHA256 digests (aka hashes or checksums) are a
+complex topic. In the context of our work at Truss, the topic is especially
+relevant when figuring out which value to use when referencing an image in an
+Amazon Elastic Container Service (ECS) task definition.
+
+### Problem Statement and Summary
+
+An [ECS task
+definition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_image)
+can pull an image by tag or digest, so which should you use? Image tags are
+potentially mutable and are not tied to the image contents. For instance, a new
+image could be created with different contents, but could reuse the same tag as
+an earlier, different image. If your ECS task definition pulls the image based
+on tag, you could be pulling in a different image than expected. An image's
+digest, on the other hand, is a hash based on the image contents. Pulling the
+image based on digest provides greater security and protection against
+unintended changes.
```

Comment:
You can configure ECR with [Immutability](https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-tag-mutability.html) so you cannot overwrite a tag.  IMO, we should always enable that.  That does make it pretty impossible do things like having an up to date `dev`, `production` or `latest` tag.  Again, IMO, you shouldn't be using those anyway.